main.py

```diff
@@ -575,6 +575,36 @@ def delete_tag_range(self, request: Request) -> JSONResponse:
             raise HTTPException(400, detail=detail)
         return JSONResponse("Operation Successful", status_code=200)
 
+    @rest('v3/interfaces/tag_ranges', methods=['GET'])
+    @validate_openapi(spec)
+    def get_all_tag_ranges(self, _):
+        """Get all tag_ranges and available_tags from interfaces"""
+        result = {}
+        for switch_id, switch in self.controller.switches.items():
```

Comment:
This iteration here isn't thread safe, since `self.controller.switches` is a shared mutable dict and you have a threaded request context, then the dict size might change mid iteration. You'll need a shallow copy to avoid an issue. Check out [an example](https://github.com/kytos-ng/topology/blob/master/main.py#L96) on topology for instance.

Reply:
Added `copy()`